corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js

```diff
@@ -143,10 +143,15 @@ hqDefine("cloudcare/js/formplayer/menus/util", function () {
                 if (searchText) {
                     event = "Searched Case List";
                 }
-                hqImport('analytix/js/kissmetrix').track.event(event, {
+                var eventData = {
                     domain: FormplayerFrontend.getChannel().request("currentUser").domain,
                     name: eventData,
-                });
+                };
+                var fields = _.pick(Util.getCurrentQueryInputs(), function (v) { return !!v; });
+                if (_.size(fields)) {
```

Comment:
nit: i just find http://underscorejs.org/#isEmpty a little nicer and explicit.

Reply:
though is the check necessary? if empty this would pass an empty fields, is that an issue?

Reply:
Agreed `isEmpty` is clearer, updated in 380c85bc021

The project team wants to know what fields users are searching by, so they don't want to know about empty fields. Empty fields shouldn't show up because ultimately this data is coming from [getAnswers](https://github.com/dimagi/commcare-hq/blob/8534c0f42e77a7c07f73f561ac3f5b6674c6a99c/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js#L235-L245), this is just a defensive check.

Reply:
thanks. Till we have underscore :)

Cool, thanks for the response.